client/src/app/site/agenda/components/topic-detail/topic-detail.component.ts

```diff
@@ -0,0 +1,171 @@
+        private promptService: PromptService
+    ) {
+        this.getTopicByUrl();
+        this.createForm();
+    }
+
+    /**
+     * Set the edit mode to the given Status
+     * @param mode
+     */
+    public setEditMode(mode: boolean): void {
+        this.editTopic = mode;
+        if (mode) {
+            this.patchForm();
+        }
+        if (!mode && this.newTopic) {
+            this.router.navigate(['./agenda/']);
+        }
+    }
+
+    /**
+     * Save a new topic as agenda item
+     */
+    public async saveTopic(): Promise<void> {
+        if (this.newTopic && this.topicForm.valid) {
+            const response = await this.repo.create(this.topicForm.value);
+            this.router.navigate([`/agenda/topics/${response.id}`]);
+            // after creating a new topic, go "back" to agenda list view
+            this.location.replaceState('/agenda/');
+        } else {
+            await this.repo.update(this.topicForm.value, this.topic);
+            this.setEditMode(false);
+        }
+    }
+
+    /**
+     * Setup the form to create or alter the topic
+     */
+    public createForm(): void {
+        this.topicForm = this.formBuilder.group({
+            title: ['', Validators.required],
+            text: ['']
+        });
+    }
+
+    /**
+     * Overwrite form Values with values from the topic
+     */
+    public patchForm(): void {
+        const topicPatch = {};
+        Object.keys(this.topicForm.controls).forEach(ctrl => {
+            topicPatch[ctrl] = this.topic[ctrl];
+        });
+        this.topicForm.patchValue(topicPatch);
+    }
+
+    /**
+     * Determine whether a new topic should be created or an existing one should
+     * be loaded using the ID from the URL
+     */
+    public getTopicByUrl(): void {
+        if (this.route.snapshot.url[1] && this.route.snapshot.url[1].path === 'new') {
+            // creates a new topic
+            this.newTopic = true;
+            this.editTopic = true;
+            this.topic = new ViewTopic();
+        } else {
+            // load existing topic
+            this.route.params.subscribe(params => {
+                this.loadTopic(params.id);
+            });
+        }
+    }
+
+    /**
+     * Loads a top from the repository
+     * @param id the id of the required topic
+     */
+    public loadTopic(id: number): void {
+        this.repo.getViewModelObservable(id).subscribe(newViewTopic => {
+            // repo sometimes delivers undefined values
+            // also ensures edition cannot be interrupted by autoupdate
+            if (newViewTopic && !this.editTopic) {
+                this.topic = newViewTopic;
+                // personalInfoForm is undefined during 'new' and directly after reloading
+                if (this.topicForm) {
+                    this.patchForm();
+                }
+            }
+        });
+    }
+
+    /**
+     * Create the absolute path to the corresponding list of speakers
```

Comment:
proper `@returns` is missing in the documentation

Reply:
yep